examples/platforms/cc2538/radio.c

```diff
@@ -649,12 +661,21 @@ void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
     }
 }
 
-void readFrame(otInstance *aInstance)
+static void readFrame(void)
 {
     uint8_t length;
     uint8_t crcCorr;
     int     i;
 
+    if (sReceiveFrame.mLength != 0)
+    {
+        /*
+         * There is already a frame present in the buffer, return early so
+         * we do not overwrite it (hopefully we'll catch it on the next run).
+         */
+        return;
```

Comment:
style: we prefer to have a single `return` statement. Can we use `otEXPECT()` here instead?